vendor/github.com/cilium/ebpf/btf/marshal.go

```diff
@@ -0,0 +1,422 @@
+package btf
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/cilium/ebpf/internal"
+)
+
+type encoderOptions struct {
+	ByteOrder binary.ByteOrder
+	// Remove function linkage information for compatibility with <5.6 kernels.
+	StripFuncLinkage bool
+}
+
+// kernelEncoderOptions will generate BTF suitable for the current kernel.
+var kernelEncoderOptions encoderOptions
+
+func init() {
+	kernelEncoderOptions = encoderOptions{
+		ByteOrder:        internal.NativeEndian,
+		StripFuncLinkage: haveFuncLinkage() != nil,
```

Comment:
this looks like it's adding unconditional setup on in an `init()` function for anything that links this package... is that new? if so, what is the cost of this, and are there any possible negative effects for binaries that don't actually use ebpf but link in this package?

Reply:
This was added as part of BTF marshalling that was added in v0.10.0. The init function was later removed in commits after the tag creation.

Reply:
Even in latest version of ebpf package, instead of `init()`, it has been moved to variables of package scope . Ref : https://github.com/cilium/ebpf/blob/master/btf/marshal.go#L20. This would still involve invoking the func.